### PR TITLE
Migrate to Rust Edition 2024

### LINKED ISSUE #209: Migration to Rust Edition 2024
As Rust version 1.85.0 enters nightly channel, it's now possible for ArceOS to migrate to Rust Edition 2024.

### PR DESCRIPTION
**Pull request by aarkegz**
Resolves #209.

### CODE REVIEW

File: Cargo.lock
Comment (equation314) [inline Cargo.lock:1]: Update percpu & allocator

File: Cargo.toml [hunk 1]
```diff
@@ -32,6 +32,7 @@ members = [
 
 [workspace.package]
 version = "0.1.0"
+edition = "2021"
```
Comment (equation314) [inline Cargo.toml]: 2024?

Comment (aarkegz) [reply Cargo.toml]: 🙃mb

File: ulib/axlibc/c/printf.h [hunk 2]
```diff
@@ -42,7 +42,7 @@
 #ifdef __cplusplus
 #include <cstdarg>
 #include <cstddef>
-extern "C" {
+unsafe extern "C" {
```
Comment (equation314) [inline ulib/axlibc/c/printf.h]: Does C code need unsafe？😂

Comment (aarkegz) [reply ulib/axlibc/c/printf.h]: `extern "C"` now requires `unsafe`

Comment (aarkegz) [reply ulib/axlibc/c/printf.h]: 😔 Sadly I just searched and replaced them, without limiting to rust files

File: ulib/axlibc/src/strftime.rs [hunk 3]
```diff
@@ -97,6 +97,14 @@ impl<T: Write> Write for CountingWriter<T> {
...
 
+#[inline]
+fn c_char_as_u8(c: c_char) -> u8 {
+    #[allow(clippy::unnecessary_cast)]
+    {
+        c as u8
+    }
+}
+
 unsafe fn strftime_inner<W: WriteByte>(
```
Comment (equation314) [inline ulib/axlibc/src/strftime.rs]: Add `allow(unnecessary_cast)` here and remove `c_char_as_u8`

**Review by equation314 — APPROVED** [review]

**Pull request merged**